### PR TITLE
new zms api to delete pending members

### PR DESCRIPTION
**Pull request by havetisyan**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java [hunk 1]
```diff
@@ -1230,7 +1230,54 @@ void executeDeleteMembership(ResourceContext ctx, String domainName, String role
...
+    void executeDeletePendingMembership(ResourceContext ctx, String domainName, String roleName,
+                String normalizedMember, String auditRef, String caller) {
+
+        // our exception handling code does the check for retry count
+        // and throws the exception it had received when the retry
+        // count reaches 0
+
+        for (int retryCount = defaultRetryCount; ; retryCount--) {
+
+            try (ObjectStoreConnection con = store.getConnection(true, true)) {
```
Comment (abvaidya) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java]: should this be autoCommit=false?

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java [hunk 2]
```diff
@@ -3282,6 +3282,56 @@ void sendMembershipApprovalNotification(final String domain, final String org, f
...
+        // policy, service, etc name)
+
+        domainName = domainName.toLowerCase();
+        roleName = roleName.toLowerCase();
+        memberName = normalizeDomainAliasUser(memberName.toLowerCase());
+
+        final String principalDomain = getPrincipalDomain(ctx);
+        metric.increment(ZMSConsts.HTTP_REQUEST, domainName, principalDomain);
+        metric.increment(caller, domainName, principalDomain);
+        Object timerMetric = metric.startTiming("deletemembership_timing", domainName, principalDomain);
```
Comment (abvaidya) [inline servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java]: should this be deletependingmembership_timing?

File: servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java (hunk 1 above)
Comment (havetisyan) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/DBService.java]: For single operations, we have auto commit enabled.

File: servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java (hunk 2 above)
Comment (havetisyan) [reply servers/zms/src/main/java/com/yahoo/athenz/zms/ZMSImpl.java]: fixed.

**Pull request merged**